source/dynamical_systems/src/PointAttractor.cpp

```diff
@@ -145,22 +143,24 @@ void PointAttractor<CartesianState>::set_base_frame(const CartesianState& base_f
 template<>
 void PointAttractor<CartesianState>::validate_and_set_parameter(const std::shared_ptr<ParameterInterface>& parameter) {
   if (parameter->get_name() == "attractor") {
-    this->set_attractor(std::static_pointer_cast<Parameter<CartesianPose>>(parameter)->get_value());
+    this->set_attractor(parameter->get_parameter_value<CartesianState>());
```

Comment:
Because of the dynamic casting in `get_parameter_value<T>`, we cannot directly cast between CartesianPose and CartesianState parameters. But, if you did want to be more permissive / less restrictive when setting the attractor parameter, it can be achieved here in the validation step, similar to how `set_gain` already works.
```c++
if (parameter->get_type == StateType::PARAMETER_CARTESIANPOSE) {
  this->set_attractor(parameter->get_parameter_value<CartesianPose>());
} else ...
```

Reply:
I think this is fine like that, we just do it in the validation step like in the `set_gain`

Reply:
Are you saying it's fine how it is now, or that you would like me to add this extra check here for more flexible usage?

Reply:
Sorry I was not doing a proper review but now I'm here. I'd actually appreciate if we could set the attractor from a CartesianPose parameter